sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpPipelinePolicy.java

```diff
@@ -22,4 +22,20 @@ public interface HttpPipelinePolicy {
      * @return A publisher that initiates the request upon subscription and emits a response on completion.
      */
     Response<?> process(HttpRequest httpRequest, HttpPipelineNextPolicy next);
+
+    /**
+     * Gets the position in the {@link HttpPipelineBuilder} the policy will be placed when added.
+     * <p>
+     * Policy order does not need to be unique. When multiple polices with the same {@link HttpPipelineOrder} are added
+     * they will be handled based on the documentation of {@link HttpPipelineOrder}.
+     * <p>
+     * By default, this method returns {@link HttpPipelineOrder#BETWEEN_RETRY_AND_AUTHENTICATION}.
+     * <p>
+     * If this method returns null, an exception will be thrown when it is added to the {@link HttpPipelineBuilder}.
+     *
+     * @return The order of this policy.
+     */
+    default HttpPipelineOrder getOrder() {
+        return HttpPipelineOrder.BETWEEN_RETRY_AND_AUTHENTICATION;
```

Comment:
why not HttpPipelineOrder.fromValue(0) (i.e. per call)? 

many policies like user-agent, request-id, headers don't need to run on every try (as long as retry policy copies headers over).

It's not blocking for this PR.

Reply:
Easier upgrade story from the current handling of `azure-core`'s `HttpPipelinePoistion` where the default used it `PER_RETRY`. We can discuss further whether we want to change this.